### PR TITLE
roachtest/cdc: fix race in latency verifier in initial_scan='only' tests

### LINKED ISSUE #111495: roachtest: cdc/initial-scan-only/parquet failed
roachtest.cdc/initial-scan-only/parquet [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11980154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11980154?buildTab=artifacts#/cdc/initial-scan-only/parquet) on master @ [27e7c166b5591b0e05bb2fd23780309b490d7af9](https://github.com/cockroachdb/cockroach/commits/27e7c166b5591b0e05bb2fd23780309b490d7af9):

```
(latency_verifier.go:185).assertValid: initial scan did not complete
(monitor.go:153).Wait: monitor failure: monitor user task failed: t.Fatal() was called
test artifacts and logs in: /artifacts/cdc/initial-scan-only/parquet/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=16</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-11980154/cdc-initial-scan-only-parquet/1695983545461/1695984352086)

</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*cdc/initial-scan-only/parquet.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-31932

### PR DESCRIPTION
**Pull request by jayshrivastava**
Before this change, it was possible for the test to finish before
the latency verifier sees the final highwater of a changefeed.
The latency verifier needs to see the final highwater in initial_scan='only'
tests to pass verification. This change makes sure that the initial_scan='only'
tests wait for the verifier to finish before tearing down the test.

Release note: None
Epic: None
Closes: https://github.com/cockroachdb/cockroach/issues/111495

### CODE REVIEW

File: pkg/ccl/changefeedccl/changefeed_processors.go [hunk 1]
```diff
@@ -1389,9 +1386,6 @@ func (cf *changeFrontier) forwardFrontier(resolved jobspb.ResolvedSpan) error {
 		return err
 	}
 
-	if log.V(1) {
-		log.Infof(cf.Ctx(), "forwarding frontier. changed: %t, resolved: %s", frontierChanged, resolved)
```
Comment (srosenberg) [inline pkg/ccl/changefeedccl/changefeed_processors.go:1393]: Curious why the infos are no longer useful? They were probably instrumented for the sake of finding this race, but perhaps they might be useful for finding other regressions?

**Review by srosenberg — APPROVED** [review]

File: pkg/ccl/changefeedccl/changefeed_processors.go (hunk 1 above)
Comment (jayshrivastava) [reply pkg/ccl/changefeedccl/changefeed_processors.go:1393]: The logging I added is a bit messy. I think I'll add neater ones later.

Comment (jayshrivastava) [reply pkg/ccl/changefeedccl/changefeed_processors.go:1393]: TYFR

**Review by miretskiy — APPROVED** [review]

**Comment by jayshrivastava** [comment]
blathers backport 23.2

**Pull request merged**